x-pack/plugins/security_solution/public/resolver/store/helpers.ts

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { Draft } from 'immer';
+import produce from 'immer';
+import type { Reducer, AnyAction } from 'redux';
+import type { ActionCreator } from 'typescript-fsa';
+import type { ReducerBuilder } from 'typescript-fsa-reducers';
+import type { ResolverState, AnalyzerState } from '../types';
+import { scaleToZoom } from './camera/scale_to_zoom';
+import { analyzerReducer } from './reducer';
+
+export const EMPTY_RESOLVER: ResolverState = {
+  data: {
+    currentRelatedEvent: {
+      loading: false,
+      data: null,
+    },
+    tree: {},
+    resolverComponentInstanceID: undefined,
+    indices: [],
+    detectedBounds: undefined,
+  },
+  camera: {
+    scalingFactor: scaleToZoom(1), // Defaulted to 1 to 1 scale
+    rasterSize: [0, 0],
+    translationNotCountingCurrentPanning: [0, 0],
+    latestFocusedWorldCoordinates: null,
+    animation: undefined,
+    panning: undefined,
+  },
+  ui: {
+    ariaActiveDescendant: null,
+    selectedNode: null,
+  },
+};
+
+export function immerCase<S, P>(
```

Comment:
this is neat!

Ignore if you feel not necessary, I just though that a method reused in many places could benefit from a small documentation, with a link to this for example https://immerjs.github.io/immer/produce (just to help those like me that aren't super familiar with `immerjs`)